source/common/network/listener_filter_buffer_impl.cc

```diff
@@ -0,0 +1,70 @@
+#include "source/common/network/listener_filter_buffer_impl.h"
+
+#include <string.h>
+
+namespace Envoy {
+namespace Network {
+
+uint64_t ListenerFilterBufferImpl::copyOut(Buffer::Instance& buffer, uint64_t length) {
+  auto size = std::min(length, data_size_);
+  if (size == 0) {
+    return 0;
+  }
+  buffer.drain(buffer.length());
+  uint8_t* raw_mem = static_cast<uint8_t*>(buffer_->frontSlice().mem_);
```

Comment:
Can you assume that all of the data is in a single slice? Potentially not? Should this be provided by the buffer impl itself? I guess maybe you gaurantee this because you have reserved a single slice. Can you add more comments and asserts?

Reply:
Yes, I reserve single slice, so I can assume that.